### ISSUE TITLE
Minification refactors JS logic and moves variable inside block scope

### ISSUE THREAD

**Issue by ianleeder** [Shazwazza/Smidge#185]
_This is from Umbraco 10.4.0.  Smidge DLLs are tagged v4.2.0.  Using option `BundlingOptions.OptimizedAndComposite`.  I'm making a fairly large assumption that this is a Smidge issue.  If for some reason Umbraco is modifying the JS, my apologies._

I created a reproducible example.  Here is my `Test.js`:
```js
document.addEventListener("DOMContentLoaded", () => {
  const body = document.querySelector("body");
  if (!body) {
    return;
  }
  const foo = 'bar'
  doIt();
  function doIt() {
      console.log(foo);
  }
});
```

After this file has been run through minification, it ends up like so:
```js
document.addEventListener("DOMContentLoaded", ()=>{
    function doIt() {
        console.log(foo)
    }
    const body = document.querySelector("body");
    if (body) {
        const foo = "bar";
        doIt()
    }
}
);
```

Notably it inverts the guard clause, and moves variables inside the block scope.  Now the function `doIt()` throws `Uncaught ReferenceError: foo is not defined`

**Comment by Shazwazza** [comment]
Hi, Smidge uses Nuglify for minification so that is an issue with Nuglify. Be sure to use the latest Nuglify version since perhaps it's already fixed, else you can log an issue on their repo.

**Comment by ianleeder** [comment]
FYI this was just fixed in Nuglify 1.22.
- I raised [issue 357](https://github.com/jbest84/NUglify/issues/357) in 2023.
- It was just fixed yesterday
- It was included in [version 1.22](https://github.com/jbest84/NUglify/releases/tag/v1.22.0)
  - It doesn't show in the summary, I think there was no PR for it.
  - Commit [5653786](https://github.com/jbest84/NUglify/commit/56537861790d8a5da02627e77933b5032e25ea86) says it fixes issue 357, and this commit shows in the [full changelog comparison page](https://github.com/jbest84/NUglify/compare/v1.21.18...v1.22.0)

I'm not sure I agree with closing this issue in Smidge, since it's still an issue here until Smidge is updated to latest Nuglify. But I haven't used it in 3 years, so it doesn't worry me too much.
